packages/barcode-scanning/ios/Plugin/BarcodeScannerHelper.swift

```diff
@@ -12,7 +12,7 @@ public class BarcodeScannerHelper {
         let screenSize: CGRect = UIScreen.main.bounds
         let imageWidth = imageSize.width
         let imageHeight = imageSize.height
-        let isPortrait = UIDevice.current.orientation == .portrait || UIDevice.current.orientation == .portraitUpsideDown
+        let isPortrait = UIDevice.current.orientation.isPortrait || UIDevice.current.orientation.isFlat
```

Comment:
Couldn't `isFlat` also be `true` if the device was in a landscape orientation before so that the UI is still in landscape mode or am I misunderstanding `isFlat`?

Reply:
ye, that's why I'm uncomfortable with this 'solution' - works for my case (I only support portrait mode UI), but then breaks the landscape case. The method for figuring out actual UI orientation seems to need to run on the main UI thread (I'm not quite sure how to sample that here, seeing as this is my first time ever touching swift 🙈 )

An alternative not-really-a-solution is to send through the orientation and let the underlying app decide how to correct, but that seems iffy.

Reply:
I looked around, and the most accurate thing I can see is to bubble it through from the `interfaceOrientation` in `layoutSubviews` https://github.com/capawesome-team/capacitor-mlkit/pull/116/files#diff-8b77d4c809bf757f8fef70caf790b8b021e201d15987c2d724bf0af350b14b97R113

This works correctly in testing on all orientations now, except that I needed to add `1-` on the landscape coords for it to map correctly in my testing - I'm not sure why this is?